eng/scripts/generate-specialized-test-projects-list.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# This script generates an MSBuild props file listing test projects that contain tests
+# marked with a specific attribute (e.g., QuarantinedTest).
+#
+# Usage: ./generate-specialized-test-projects-list.sh <attribute_name>
+# Example: ./generate-specialized-test-projects-list.sh QuarantinedTest
+
+set -euo pipefail
+set -x
+
+if [[ $# -ne 2 ]]; then
+    echo "Usage: $0 <attribute_name> <output_file>"
+    exit 1
+fi
+
+ATTRIBUTE_NAME="${1:?Usage: $0 <attribute_name>}"
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+OUTPUT_FILE="$2"
+
+mkdir -p `dirname $OUTPUT_FILE`
+
+# Find all test files with the attribute and extract unique top-level test directories
+PROJECTS=$(grep -rl "^ *\[${ATTRIBUTE_NAME}(\"[^\"]*\")\]" "$REPO_ROOT/tests" 2>/dev/null \
```

Comment:
This is kind of gross. Why can't we use normal Test traits / categories / etc? Test frameworks have this stuff built in.